src/IECoreGL/ColorTexture.cpp

```diff
@@ -58,7 +58,7 @@ ColorTexture::ColorTexture( unsigned int width, unsigned int height )
 	glTexParameteri( GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_NEAREST );
 	glTexParameteri( GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_NEAREST );
 
-	glTexImage2D( GL_TEXTURE_2D, 0, GL_RGBA, width, height, 0, GL_RGBA,
+	glTexImage2D( GL_TEXTURE_2D, 0, GL_RGBA32F, width, height, 0, GL_RGBA,
```

Comment:
I changed this in order to use the existing `ColorTexture` class for the camera-depth texture. Without making this `GL_RGBA32F`, values were getting clamped to [0.0, 1.0]. I think it does represent a potentially significant increase in buffer storage cost though, so maybe it's preferable to add a `FloatTexture` class, or somehow piggy back on `DepthBuffer`?

https://www.khronos.org/opengl/wiki/Image_Format states that `The bitdepth can also be omitted as well, but only with unsigned normalized formats. Doing so gives OpenGL the freedom to pick a bitdepth.` The highest bit depth for the unsigned normalized floats we were defaulting to with `GL_RGBA` is 16, so at the least this change doubles the memory requirement.

Reply:
Yeah, I think we should avoid increasing the overhead for other textures. Adding a FloatTexture would seem pretty logical given the current design, but I don't think I'd design things this way again. Maybe the simplest is just to add an option to the ColorTexture constructor to pass the internal format we want?

Reply:
I think https://github.com/ImageEngine/cortex/pull/1404/commits/5ae6f75f163ae9468c05152397ee290dbea3753c does the job without breaking API.